diesel/src/connection/mod.rs

```diff
@@ -185,8 +193,57 @@ pub trait Connection: SimpleConnection + Sized + Send {
     #[doc(hidden)]
     fn execute_returning_count<T>(&self, source: &T) -> QueryResult<usize>
     where
+        Self: Sized,
         T: QueryFragment<Self::Backend> + QueryId;
 
     #[doc(hidden)]
-    fn transaction_manager(&self) -> &Self::TransactionManager;
+    fn transaction_manager(&self) -> &dyn TransactionManager<Self>
+    where
+        Self: Sized;
+}
+
+/// A variant of the [`Connection`](trait.Connection.html) that is
+/// usable with dynamic dispatch
+///
+/// If you are looking for a way to use pass database connections
+/// for different database backends around in your application
+/// this trait won't help you much. Normally you should only
+/// need to use this trait if you are interacting with a connection
+/// passed to a [`Migration`](../migration/trait.Migration.html)
+pub trait BoxableConnection<DB: Backend>: Connection<Backend = DB> + std::any::Any {
+    #[doc(hidden)]
+    fn as_any(&self) -> &dyn std::any::Any;
+}
+
+impl<C> BoxableConnection<C::Backend> for C
+where
+    C: Connection + std::any::Any,
+{
+    fn as_any(&self) -> &dyn std::any::Any {
+        self
+    }
+}
+
+impl<DB: Backend> dyn BoxableConnection<DB> {
```

Comment:
It is no really a special syntax, as this just implements some methods for a type as any other impl block. The only difference to "normal" impl blocks is that the type is `dyn Trait` here instead of a `Sized` type. In the end this allows us to have functions with generic arguments on a trait object (`dyn Trait`).